zenoh/src/prelude.rs

```diff
@@ -23,1155 +23,1220 @@
-    pub(crate) fn empty() -> Self {
-        SourceInfo {
-            source_id: None,
-            source_sn: None,
-            first_router_id: None,
-            first_router_sn: None,
+    /// Informations on the source of a zenoh [`Sample`].
+    #[derive(Debug, Clone)]
+    pub struct SourceInfo {
+        /// The [`ZenohId`] of the zenoh instance that published the concerned [`Sample`].
+        pub source_id: Option<ZenohId>,
+        /// The sequence number of the [`Sample`] from the source.
+        pub source_sn: Option<ZInt>,
+        /// The [`ZenohId`] of the first zenoh router that routed this [`Sample`].
+        pub first_router_id: Option<ZenohId>,
+        /// The sequence number of the [`Sample`] from the first zenoh router that routed it.
+        pub first_router_sn: Option<ZInt>,
+    }
+
+    impl SourceInfo {
+        pub(crate) fn empty() -> Self {
+            SourceInfo {
+                source_id: None,
+                source_sn: None,
+                first_router_id: None,
+                first_router_sn: None,
+            }
         }
     }
-}
 
-impl From<DataInfo> for SourceInfo {
-    fn from(data_info: DataInfo) -> Self {
-        SourceInfo {
-            source_id: data_info.source_id,
-            source_sn: data_info.source_sn,
-            first_router_id: data_info.first_router_id,
-            first_router_sn: data_info.first_router_sn,
+    impl From<DataInfo> for SourceInfo {
+        fn from(data_info: DataInfo) -> Self {
+            SourceInfo {
+                source_id: data_info.source_id,
+                source_sn: data_info.source_sn,
+                first_router_id: data_info.first_router_id,
+                first_router_sn: data_info.first_router_sn,
+            }
         }
     }
-}
 
-impl From<Option<DataInfo>> for SourceInfo {
-    fn from(data_info: Option<DataInfo>) -> Self {
-        match data_info {
-            Some(data_info) => data_info.into(),
-            None => SourceInfo::empty(),
+    impl From<Option<DataInfo>> for SourceInfo {
+        fn from(data_info: Option<DataInfo>) -> Self {
+            match data_info {
+                Some(data_info) => data_info.into(),
+                None => SourceInfo::empty(),
+            }
         }
     }
-}
 
-/// The kind of a [`Sample`].
-#[derive(Debug, Copy, Clone, PartialEq)]
-pub enum SampleKind {
-    /// if the [`Sample`] was caused by a `put` operation.
-    Put = data_kind::PUT as isize,
-    /// if the [`Sample`] was caused by a `patch` operation.
-    Patch = data_kind::PATCH as isize,
-    /// if the [`Sample`] was caused by a `delete` operation.
-    Delete = data_kind::DELETE as isize,
-}
+    /// The kind of a [`Sample`].
+    #[derive(Debug, Copy, Clone, PartialEq)]
+    pub enum SampleKind {
+        /// if the [`Sample`] was caused by a `put` operation.
+        Put = data_kind::PUT as isize,
+        /// if the [`Sample`] was caused by a `patch` operation.
+        Patch = data_kind::PATCH as isize,
```

Comment:
There is no `Patch` operation implemented as of today. Let's remove it from this enum as well from the `data_kind`.

Reply:
I'm kind of on the fence on that: if someone wants to implement a queryable that supports path operations, wouldn't it be nice for them to have this?

Just because we don't support it anywhere yet doesn't mean it doesn't make sense for these queries to be able to exist.

But on the other hand, I get that it would be confusing for users that first-party queryables don't support this operation despite it existing, because it is indeed a pointless operation for now.